var/spack/repos/builtin/packages/pythia8/package.py

```diff
@@ -23,13 +23,32 @@ class Pythia8(AutotoolsPackage):
     version('8212', sha256='f8fb4341c7e8a8be3347eb26b00329a388ccf925313cfbdba655a08d7fd5a70e')
 
     variant('shared', default=True, description='Build shared library')
+    variant('hepmc', default=True, description='Build HepMC2 extensions')
+    variant('evtgen', default=False, description='Build EvtGen extensions')
+    variant('root', default=False, description='Build ROOT extensions')
+    variant('fastjet', default=False, description='Build fastjet extensions')
 
     depends_on('rsync', type='build')
+    depends_on('hepmc@:2.99.99', when="+hepmc")
+    depends_on('root', when="+root")
+    depends_on('evtgen', when="+evtgen")
+    depends_on("fastjet@3.0.0:", when="+fastjet")
 
     def configure_args(self):
         args = []
 
         if '+shared' in self.spec:
             args.append('--enable-shared')
+        if '+hepmc' in self.spec:
+            args.append('--with-hepmc=%s' % self.spec["hepmc"].prefix)
+        if '+fastjet' in self.spec:
+            args.append('--with-fastjet3=%s' % self.spec["fastjet"].prefix)
+        if '+evtgen' in self.spec:
+            args.append('--with-evtgen=%s' % self.spec["evtgen"].prefix)
+        if '+root' in self.spec:
+            args.append('--with-root=%s' % self.spec["root"].prefix)
```

Comment:
If you use `--without-foo` in an else statement, is it possible to make sure it doesn't enable these options?

Reply:
When I tried it, the extensions were only built if the `--with-foo` argument was explicitly present, even if `foo` was loaded. But it definitly won't hurt to add the else statement, so I added it.